// lk-api/index.js
const {
  IgStalk,
  IgStory,
  IgDl,
  Pinterest,
  TiktokDl,
  Chara,
  Otaku,
  GhStalk,
  Emoji2Img,
  ModDroid,
  PalingMurah,
  ApkMody,
  KompasNews,
  TribunNews,
  JalanTikus,
  Hoax,
  HappyMod,
  ServerMc,
  Gempa,
  KodePos,
  PlayStore,
  TebakGambar,
} = require('./lib/function')
const { Halah } = require('./lib/halah')
const { Hilih } = require('./lib/hilih')
const { Huluh } = require('./lib/huluh')
const { Heleh } = require('./lib/heleh')
const { Holoh } = require('./lib/holoh')
const { HttpHeaders } = require('./lib/headers')
const { Ip } = require('./lib/ip')
const { IpLookup } = require('./lib/iplookup')
const { JooxPlay } = require('./lib/jooxplay')
const { Proxy } = require('./lib/proxy')
const { InfoNPM } = require('./lib/infonpm')
const {
  YtdlMp3,
  YtdlMp4,
  YtPlayMp3,
  YtPlayMp4,
  YtSearch
} = require('./lib/yt')
const { Base } = require('./lib/base')
const { CovidId, CovidGlobal } = require('./lib/covid')
const { SearchRepo } = require('./lib/ghrepo')
const { OtakuSearch, Anoboy } = require('./lib/otaku')
const { Shortlink } = require('./lib/shortlink')
const { Sfw, Nsfw} = require('./lib/sfwnsfw')
const { BioskopComingSoon } = require('./lib/film')
const { NekonimeSearch } = require('./lib/animes')
const {
  DataMasjid,
  ListSurah,
  Surah,
} = require('./lib/islam')
const {
  GempaTerkini,
  InfoIklim,
  CitraSatelit,
  PrakiraanGelombang,
  PrakiraanAngin,
  PotensiKebakaran,
  CuacaPenerbangan,
} = require('./lib/bmkg')
const {
  TxtProBlueMetal2,
  TxtProThunder,
  TxtProNeonLight
} = require('./lib/textpro')

module.exports.IgStalk = IgStalk
module.exports.IgStory = IgStory
module.exports.IgDl = IgDl
module.exports.Pinterest = Pinterest
module.exports.TiktokDl = TiktokDl
module.exports.Halah = Halah
module.exports.Hilih = Hilih
module.exports.Huluh = Huluh
module.exports.Heleh = Heleh
module.exports.Holoh = Holoh
module.exports.Holoh = Holoh
module.exports.HttpHeaders = HttpHeaders
module.exports.Ip = Ip
module.exports.IpLookup = IpLookup
module.exports.Proxy = Proxy
module.exports.YtdlMp3 = YtdlMp3
module.exports.YtdlMp4 = YtdlMp4
module.exports.YtPlayMp3 = YtPlayMp3
module.exports.YtPlayMp4 = YtPlayMp4
module.exports.YtSearch = YtSearch
module.exports.Chara = Chara
module.exports.Otaku = Otaku
module.exports.GhStalk = GhStalk
module.exports.Emoji2Img = Emoji2Img
module.exports.ModDroid = ModDroid
module.exports.PalingMurah = PalingMurah
module.exports.ApkMody = ApkMody
module.exports.KompasNews = KompasNews
module.exports.TribunNews = TribunNews
module.exports.JalanTikus = JalanTikus
module.exports.Hoax = Hoax
module.exports.HappyMod = HappyMod
module.exports.ServerMc = ServerMc
module.exports.Gempa = Gempa
module.exports.KodePos = KodePos
module.exports.PlayStore = PlayStore
module.exports.TebakGambar = TebakGambar
module.exports.Base = Base
module.exports.CovidId = CovidId
module.exports.CovidGlobal = CovidGlobal
module.exports.OtakuSearch = OtakuSearch
module.exports.SearchRepo = SearchRepo
module.exports.InfoNPM = InfoNPM
module.exports.Shortlink = Shortlink
module.exports.Sfw = Sfw
module.exports.Nsfw = Nsfw
module.exports.BioskopComingSoon = BioskopComingSoon
module.exports.NekonimeSearch = NekonimeSearch
module.exports.DataMasjid = DataMasjid
module.exports.GempaTerkini = GempaTerkini
module.exports.InfoIklim = InfoIklim
module.exports.CitraSatelit = CitraSatelit
module.exports.PrakiraanGelombang = PrakiraanGelombang
module.exports.PrakiraanAngin = PrakiraanAngin
module.exports.PotensiKebakaran = PotensiKebakaran
module.exports.CuacaPenerbangan = CuacaPenerbangan
module.exports.ListSurah = ListSurah
module.exports.Surah = Surah
module.exports.TxtProThunder = TxtProThunder
module.exports.TxtProBlueMetal2 = TxtProBlueMetal2
module.exports.TxtProNeonLight = TxtProNeonLight
