// lk-api/lib/animes.js
const axios = require('axios')
const cheerio = require('cheerio')

function NekonimeSearch(search) {
  return new Promise((resolve, reject) => {
    axios.get(`https://nekonime.vip/?s=${search}`)
      .then(({
        data
      }) => {
        const $ = cheerio.load(data)
        let title = []
     let url = []
     let img = []
 	$('body > div > div.container-fluid > div > div > div.row > div.col-md-8 > div > div > article > div.article-img > a').get().map((rest) => {
         url.push($(rest).attr('href'))
         })
         $('body > div > div.container-fluid > div > div > div.row > div.col-md-8 > div > div > article > div.article-body > h3 > a').get().map((rest) => {
        	title.push($(rest).text())
         })
         $('body > div > div.container-fluid > div > div > div.row > div.col-md-8 > div > div > article > div.article-img > a > img').get().map((rest) => {
        	img.push($(rest).attr('src'))
         })
     let result = []
     for (let i = 0; i < url.length; i++) {
          result.push({
               	url: url[i],
               	title: title[i],
               	thumb: img[i]
          })
     }
            
        const res = {
            status: 200,
            coder: '@ariasu.xyz',
            result: result
          }
          resolve(res)
      })
      .catch(reject)
  })
}

module.exports.NekonimeSearch = NekonimeSearch
