// lk-api/lib/textpro.js
const zrapi = require('zrapi')

const type = {
  thunder: "https://textpro.me/online-thunder-text-effect-generator-1031.html",
  thunder2: "https://textpro.me/create-thunder-text-effect-online-881.html",
  neon_light: "https://textpro.me/create-3d-neon-light-text-effect-online-1028.html",
  neon_light2: "https://textpro.me/neon-light-text-effect-online-882.html",
  glitch: "https://textpro.me/create-impressive-glitch-text-effects-online-1027.html",
  glitch2: "https://textpro.me/create-a-glitch-text-effect-online-free-1026.html",
  glitch3: "https://textpro.me/create-glitch-text-effect-style-tik-tok-983.html",
  devil_wings: "https://textpro.me/create-neon-devil-wings-text-effect-online-free-1014.html",
  technology_neon: "https://textpro.me/create-a-futuristic-technology-neon-light-text-effect-1006.html",
  vintage_neon: "https://textpro.me/create-realistic-vintage-style-light-bulb-1000.html",
  galaxy_style: "https://textpro.me/neon-light-text-effect-with-galaxy-style-981.html",
  holographic: "https://textpro.me/holographic-3d-text-effect-975.html",
  neon_text: "https://textpro.me/neon-text-effect-online-963.html",
  neon_text2: "https://textpro.me/neon-text-effect-online-879.html",
  happy_new_year: "https://textpro.me/happ-new-year-card-firework-gif-959.html",
  firework_sparkle: "https://textpro.me/firework-sparkle-text-effect-930.html",
  rainbow_equalizer: "https://textpro.me/rainbow-equalizer-text-effect-902.html",
  matrix_style: "https://textpro.me/matrix-style-text-effect-online-884.html",
  bokeh_text: "https://textpro.me/bokeh-text-effect-876.html",
  green_neon: "https://textpro.me/green-neon-text-effect-874.html",
  advanced_glow: "https://textpro.me/free-advanced-glow-text-effect-873.html",
  cinematic_horror: "https://textpro.me/create-a-cinematic-horror-text-effect-1045.html",
  metallic_text: "https://textpro.me/create-a-metallic-text-effect-free-online-1041.html",
  transformers: "https://textpro.me/create-a-transformer-text-effect-online-1035.html",
  harry_potter: "https://textpro.me/create-harry-potter-text-effect-online-1025.html",
  glossy_metal: "https://textpro.me/create-a-3d-glossy-metal-text-effect-1019.html",
  dark_gold: "https://textpro.me/metal-dark-gold-text-effect-984.html",
  metal_purple: "https://textpro.me/metal-purple-dual-effect-973.html",
  deluxe_silver: "https://textpro.me/deluxe-silver-text-effect-970.html",
  blue_metal: "https://textpro.me/glossy-blue-metal-text-effect-967.html",
  blue_metal2: "https://textpro.me/blue-metal-text-effect-831.html",
  deluxe_gold: "https://textpro.me/deluxe-gold-text-effect-966.html",
  metal_galaxy: "https://textpro.me/text-logo-3d-metal-galaxy-943.html",
  steel_text: "https://textpro.me/steel-text-effect-online-921.html",
  rusty_metal: "https://textpro.me/rusty-metal-text-effect-860.html",
  metal_rainbow: "https://textpro.me/metal-rainbow-text-effect-854.html",
  shiny_metal: "https://textpro.me/shiny-metal-text-effect-852.html",
  hot_metal: "https://textpro.me/hot-metal-text-effect-843.html",
  eroded_metal: "https://textpro.me/eroded-metal-text-effect-834.html",
}

function TxtProBlueMetal2(text) {
	return new Promise((resolve, reject) => {
	  zrapi 
    .textpro(type.blue_metal2, [ text ])
    .then((url) => {
      const result = {
        status_code: 200,
        coder: '@ariasu.xyz',
        date: Date(),
        result: {
          text: text,
          scraping: type.blue_metal2,
          output: url
        }
      }
      resolve(result)
    })
    .catch(reject)
	})
}
function TxtProThunder(text) {
	return new Promise((resolve, reject) => {
	  zrapi 
    .textpro(type.thunder, [ text ])
    .then((url) => {
      const result = {
        status_code: 200,
        coder: '@ariasu.xyz',
        date: Date(),
        result: {
          text: text,
          scraping: type.thunder,
          output: url
        }
      }
      resolve(result)
    })
    .catch(reject)
	})
}
function TxtProNeonLight(text) {
	return new Promise((resolve, reject) => {
	  zrapi 
    .textpro(type.neon_light, [ text ])
    .then((url) => {
      const result = {
        status_code: 200,
        coder: '@ariasu.xyz',
        date: Date(),
        result: {
          text: text,
          scraping: type.neon_light,
          output: url
        }
      }
      resolve(result)
    })
    .catch(reject)
	})
}

module.exports.TxtProBlueMetal2 = TxtProBlueMetal2
module.exports.TxtProThunder = TxtProThunder
module.exports.TxtProNeonLight = TxtProNeonLight
